### PR TITLE
Added functionality to persist the number of parking available.

### PR DESCRIPTION
**Pull request by AndresConfortiGL**
## Description
Third PR - Added functionality to persist the number of parking available.

## Story Type
- New feature :white_check_mark:
- Change request
- Bug fix
- Tech

## How Can This Be Tested?
By pressing the button, you will see the text above it display the amount of parking available.

## Evidences
| Nexus S | Pixel 3a |
| ------------- | ------------- |
| nexus | ezgif-5-54c856488c |

## Tested on
The project was tested on Android 8.0 / API 26 using the following devices:

- Nexus S
- Pixel 3a

## Package Structure
| Screenshot |
| ------------- |
| structure1 |

### CODE REVIEW

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java [hunk 1]
```diff
@@ -4,4 +4,15 @@
 
 public class ParkingModel implements ParkingContract.Model {
 
+    private int AvailableParkingSpots = 0;
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java]: Use an invalid value as default, for example -1.

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenter.java [hunk 2]
```diff
@@ -11,4 +11,11 @@ public ParkingPresenter(ParkingContract.Model parkingModel, ParkingContract.View
         this.parkingModel = parkingModel;
         this.parkingView = parkingView;
     }
+
+    @Override
+    public void showAvailableParkingSpots(){
+        parkingModel.setAvailableParkingSpots(999);
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenter.java]: Don't use hardcodes. You can save this in a constant

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/ParkingView.java [hunk 3]
```diff
@@ -7,7 +7,14 @@
...
 
+    private ActivityMainBinding binding;
+
     public ParkingView(Activity activity, ActivityMainBinding binding) {
         super(activity);
+        this.binding = binding;
+    }
+    @Override
+    public void displayAvailableParkingSpots(int parkingSpots){
+    binding.parkingTextView.setText("AVAILABLE SPOTS: " + parkingSpots);
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/ParkingView.java]: Don't use hardcodes. This should be a string resource

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <resources>
     <string name="app_name">JavaBaseProject</string>
-    <string name="activity_main_button_text">Button</string>
-    <string name="activity_main_title_text">Parking Text</string>
+    <string name="activity_main_button_text">PRESS ME</string>
+    <string name="activity_main_title_text">- PARKING SPOTS AVAILABLE -</string>
```
Comment (Georgina-Fernandez) [inline app/src/main/res/values/strings.xml]: This should be "AVAILABLE PARKING SPOTS"

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/base/ActivityView.java [hunk 5]
```diff
@@ -15,4 +15,5 @@ public ActivityView(Activity activity) {
     public Activity getActivity() {
         return activityRef.get();
     }
+
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/base/ActivityView.java]: Erase this empty space

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenter.java (hunk 2 above)
Comment (AndresConfortiGL) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenter.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java (hunk 1 above)
Comment (AndresConfortiGL) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java]: done

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (AndresConfortiGL) [reply app/src/main/res/values/strings.xml]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/base/ActivityView.java (hunk 5 above)
Comment (AndresConfortiGL) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/base/ActivityView.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/ParkingView.java (hunk 3 above)
Comment (AndresConfortiGL) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/ParkingView.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenter.java [hunk 6]
```diff
@@ -6,9 +6,16 @@ public class ParkingPresenter implements ParkingContract.Presenter {
...
 
     public ParkingPresenter(ParkingContract.Model parkingModel, ParkingContract.View parkingView) {
         this.parkingModel = parkingModel;
         this.parkingView = parkingView;
     }
+
+    @Override
+    public void showAvailableParkingSpots(){
+        parkingModel.setAvailableParkingSpots(testValue);
+        parkingView.displayAvailableParkingSpots(String.valueOf(parkingModel.getAvailableParkingSpots()));
```
Comment (jnicolaslaurenti) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenter.java:19]: It's not the presenter's responsibility to parse the numbers, it's the model's responsibility

Comment (AndresConfortiGL) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenter.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenter.java [hunk 7]
```diff
@@ -6,9 +6,16 @@ public class ParkingPresenter implements ParkingContract.Presenter {
 
     private final ParkingContract.Model parkingModel;
     private final ParkingContract.View parkingView;
+    private static final int DEFAULT_VALUE = -1;
```
Comment (jnicolaslaurenti) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenter.java]: this variable must be in ParkingModel

Comment (AndresConfortiGL) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ParkingPresenter.java]: done

**Review by jnicolaslaurenti — APPROVED** [review]

**Review by mariano98montero — APPROVED** [review]

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java [hunk 8]
```diff
@@ -4,4 +4,17 @@
 
 public class ParkingModel implements ParkingContract.Model {
 
+    private static final int DEFAULT_VALUE = -1;
+    private int availableParkingSpots = DEFAULT_VALUE;
+
```
Comment (DiegoDiazGlobant) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java:9]: Remove extra line please

Comment (AndresConfortiGL) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java:9]: done

**Review by DiegoDiazGlobant — APPROVED** [review]

**Review by Georgina-Fernandez — APPROVED** [review]

**Pull request merged**